src/app/ui-components/btcAmountSelector.tsx

```diff
@@ -0,0 +1,236 @@
+  satsToBtc(num)
+    .toFixed(8)
+    .replace(/\.?0+$/, '');
+
+type Props = {
+  amountSats: string;
+  setAmountSats: (amount: string) => void;
+  sendMax: boolean;
+  setSendMax: (sendMax: boolean) => void;
+  disabled?: boolean;
+};
+
+function AmountSelector({
+  amountSats,
+  setAmountSats,
+  sendMax,
+  setSendMax,
+  disabled = false,
+}: Props) {
+  const { t } = useTranslation('translation', { keyPrefix: 'SEND' });
+  const { btcBalance: btcBalanceSats, btcFiatRate, fiatCurrency } = useWalletSelector();
+
+  const [amountDisplay, setAmountDisplay] = useState(
+    amountSats && satsToBtcString(new BigNumber(amountSats)),
+  );
+  const [useBtcValue, setUseBtcValue] = useState(true);
+
+  useEffect(() => {
+    if (!sendMax) return;
+
+    const amountToDisplay =
+      amountSats &&
+      (useBtcValue
+        ? satsToBtcString(new BigNumber(amountSats))
+        : getBtcFiatEquivalent(new BigNumber(amountSats), BigNumber(btcFiatRate))
+            .toNumber()
+            .toFixed(2));
+
+    if (amountToDisplay !== amountDisplay) {
+      setAmountDisplay(amountToDisplay);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps -- We specifically only want to run this on these 2 deps
+  }, [sendMax, amountSats]);
+
+  const btcBalance = new BigNumber(btcBalanceSats);
+  const balance = useBtcValue
+    ? satsToBtcString(btcBalance)
+    : getBtcFiatEquivalent(btcBalance, new BigNumber(btcFiatRate)).toFixed(2);
+
+  const sendAmountConverted = useBtcValue
+    ? getBtcFiatEquivalent(new BigNumber(amountSats), BigNumber(btcFiatRate)).toNumber().toFixed(2)
+    : satsToBtcString(new BigNumber(amountSats));
+
+  const handleAmountChange = (newAmount: string) => {
+    const isValidInput = inputValidator.test(newAmount);
+    if (!isValidInput) return;
+
+    setAmountDisplay(newAmount);
+
+    if (!newAmount) {
+      setAmountSats('0');
+      setSendMax(false);
+      return;
+    }
+
+    const isValidAmount = useBtcValue
+      ? btcInputValidator.test(newAmount)
+      : fiatInputValidator.test(newAmount);
+
+    if (!isValidAmount) return;
+
+    setSendMax(false);
+
+    if (useBtcValue) {
+      setAmountSats(btcToSats(new BigNumber(newAmount)).toString());
+    } else {
+      const btcAmount = btcToSats(
+        getFiatBtcEquivalent(new BigNumber(newAmount), new BigNumber(btcFiatRate)),
+      );
+      setAmountSats(btcAmount.toString());
+    }
+  };
+
+  const handleUseBtcValueChange = () => {
+    if (disabled) return;
+
+    const shouldUseBtcValue = !useBtcValue;
+    setUseBtcValue(shouldUseBtcValue);
+
+    if (!amountDisplay || Number.isNaN(+amountDisplay)) return;
+
+    if (shouldUseBtcValue) {
+      // convert outer sats amount to btc
+      setAmountDisplay(satsToBtcString(new BigNumber(amountSats)));
+    } else {
+      // convert btc to fiat
+      const fiatAmount = getBtcFiatEquivalent(new BigNumber(amountSats), BigNumber(btcFiatRate))
+        .toNumber()
+        .toFixed(2);
```

Comment:
what does the .toNumber() do here?

Reply:
My bad. Didn't realise BigNumber had this function. Will remove it in a few places 🙌